aa/MTV-475: adding 2.4.0 release notes (#375)
kubev2v/forklift-documentation: documentation/modules/rn-2.4.adoc

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-24_{context}"]
= {project-full} 2.4

> You can migrate virtual machines (VMs) from VMware vSphere or {rhv-full} to {virt} with {the-lc} {project-first}.

Reviewer: ```suggestion
You can migrate virtual machines (VMs) from VMware vSphere or {rhv-full} or {osp} to {virt} with {the-lc} {project-first}.
```

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-24_{context}"]
== Technical changes

> This release has the following technical changes:

Reviewer: we lost an important one that existed in #364 :
```
.Faster disk transfers by ovirt-imageio client (ovirt-img)

Disk transfers use ovirt-imageio client (ovirt-img) instead of Containerized Data Import (CDI) when migrating from RHV to the local OCP cluster, accelerating the migration.
```

Author: Added

> .Faster disk image migration without using virt-v2v

Reviewer: maybe this should better be `Faster disk image migration from RHV`?

Author: changed

> IDisk images are not converted anymore using virt-v2v when migrating from RHV. This way, the migration is faster and not limited to guest operating systems that are supported by virt-v2v. link:https://github.com/kubev2v/forklift-controller/issues/403[(forklift-controller#403)]

Reviewer: ```suggestion
Disk images are not converted anymore using virt-v2v when migrating from RHV. This way, the migration is faster and not limited to guest operating systems that are supported by virt-v2v. link:https://github.com/kubev2v/forklift-controller/issues/403[(forklift-controller#403)]
```

.Faster migration using conversion pod disk transfer

> vSphere to the local OCP cluster migration uses conversion pod instead of Containerized Data Importer (CDI), accelerating the migration.

Reviewer: ```suggestion
When migrating from vSphere to the local OCP cluster, the conversion pod transfers the disk data instead of Containerized Data Importer (CDI), accelerating the migration.
```

.Migrated virtual machines are not scheduled on the target OCP cluster

The migrated virtual machines are no longer scheduled on the target OCP cluster. This enables migrating VMs that cannot start due to limit constraints at the migration time on the target.

.StorageProfile resource needs to be updated for a non-provisioner storage class

You must update the `StorageProfile` resource with `accessModes` and `volumeMode` for non-provisioner storage classes such as NFS. The documentation includes a link to the relevant procedure.

[id="new-features-and-enhancements-24_{context}"]
== New features and enhancements

This release has the following features and improvements:

.OCP console plugin

> This version of {project-full} has replaced the stand-alone UI with a UI that is integrated in the OCP Console. The new UI operates as an OCP Console plugin. It is implemented by default in version 2.4, disabling the old UI. You can enable the old UI by setting `feature_ui: true` in ForkliftController. link:https://issues.redhat.com/browse/MTV-427[(MTV-427)]

Reviewer: copying from #364 :
```suggestion
This version of {project-full} has replaced the stand-alone UI with a UI that is integrated in the OCP Console. The new UI operates as an OCP Console plugin that adds the sub-menu `Migration` to the navigation bar. It is implemented by default in version 2.4, disabling the old UI. You can enable the old UI by setting `feature_ui: true` in ForkliftController. link:https://issues.redhat.com/browse/MTV-427[(MTV-427)]
```
and I'm not sure about the phrasing of: "It implements by default .."

Author: Change made

> .OpenStack migration

Reviewer: can we please move the OpenStack migration to the top of this section?

Author: Moved

> See new link:https://access.redhat.com/documentation/en-us/migration_toolkit_for_virtualization/2.3/html-single/installing_and_using_the_migration_toolkit_for_virtualization/index#rhv-prerequisites_mtv[prerequisites] for RHV and VMware migration, and link:https://access.redhat.com/documentation/en-us/migration_toolkit_for_virtualization/2.3/html-single/installing_and_using_the_migration_toolkit_for_virtualization/index#migrating-virtual-machines-cli_mtv[procedure] to migrate virtual machines. link:https://issues.redhat.com/browse/MTV-426[(MTV-426)]

Reviewer: the links are incorrect
shouldn't we put something more general here like "OpenStack was added as a source provider. Only cold migrations are supported as tech preview in this release"?

.Skip certification option

Skip certificate validation' option was added to VMware and RHV providers. If checked, the provider's certificate will not be validated and the UI will not ask for specifying a CA certificate.

.Only third-party certificate required

When defining a RHV provider that set with third-party certificate, only the third-party certificate needs to be specified and MTV would retrieve the Manager CA certificate automatically.

.Conversion of VMs with RHEL9 guest operating system

> VM conversion with RHEL9 in virt-v2v installs `qemu-ga` within the converted guest. link:https://issues.redhat.com/browse/MTV-332[(MTV-332)]

Reviewer: ```suggestion
"For cold migrations from vSphere to the local OCP cluster, we use virt-v2v on RHEL 9 that supports conversion of virtual machines with RHEL 9 as the guest operating system". link:https://issues.redhat.com/browse/MTV-332[(MTV-332)]
```

[id="known-issues-24_{context}"]

> == Known issues

Reviewer: we need to add a new one:
```
.VMs from vSphere with RHEL 9 guest operating system may start with network interfaces that are down

When migrating VMs that are installed with RHEL 9 as guest operating system from vSphere, their network interfaces could be disabled when they start in Openshift Virtualization
```

Author: > we need to add a new one:
> 
> ```
> .VMs from vSphere with RHEL 9 guest operating system may start with network interfaces that are down
> 
> When migrating VMs that are installed with RHEL 9 as guest operating system from vSphere, their network interfaces could be disabled when they start in Openshift Virtualization
> ```

Hey Arik, do you have a ticket number for this please?

Reviewer: checking with Nisim

.PVC deletion hangs after archiving and deleting migration plan

When archiving and deleting a migration plan that failed, its PVCs and PVs are not deleted. Workaround: remove the CDI importer pods and then remove the remaining PVCs and PVs. link:https://issues.redhat.com/browse/MTV-493[(MTV-493)]

.VM with multiple disks may boot from non-bootable disk after migration

VM with multiple disks that was migrated may not be able to boot on the target OCP cluster. +
Workaround: set the boot order appropriately to boot from the bootable disk. link:https://issues.redhat.com/browse/MTV-433[(MTV-433)]

.Non-supported guest operating systems in warm migrations

Warm migrations and migrations to remote OCP clusters from vSphere do not support all types of guest operating systems that are supported in cold migrations to the local OCP cluster. It is a consequence of using RHEL 8 in the former case and RHEL 9 in the latter case. +
See link:https://access.redhat.com/articles/1351473[Converting virtual machines from other hypervisors to KVM with virt-v2v in RHEL 7, RHEL 8, and RHEL 9] for the list of supported guest operating systems.

[id="resolved-issues-24_{context}"]
== Resolved issues

> This release has the following resolved issue:

Reviewer: please add another one:
```
.Users without cluster-admin role can create new providers
Previously, cluster-admin role was required to browse and create providers. In this release, users with sufficient permissions on MTV resources (providers, plans, migrations, networkmaps, storagemaps, hooks) can operate MTV without cluster-admin permissions.
```
issue: https://issues.redhat.com/browse/MTV-334

Author: Added

.Map IDE to SATA +

> CNV does not support IDE disk interface, so the disks are mapped to SATA. link:https://issues.redhat.com/browse/MTV-345[(MTV-345)]

Reviewer: this one can be dropped

Author: Dropped

.Improve invalid/conflicting VM name handling

> Users can automate VM renaming during mogration, to fit RFC 1123. link:https://issues.redhat.com/browse/MTV-212[(MTV-212)]

Reviewer: ```suggestion
Improve the automatic renaming of VMs during migration to fit RFC 1123. This feature that was introduced in 2.3.4 is enhanced to cover more special cases. 
link:https://issues.redhat.com/browse/MTV-212[(MTV-212)]
```

.Prevent locking user accounts due to incorrect credentials

> When a user specifies an incorrect password for RHV providers, he no longer gets locked in RHV. An error returns when the RHV manager is accessible when adding the provider. If the RHV manager is not accessible, the provider is added, but there would be not further attempt after failing, due to incorrect credentials. link:https://issues.redhat.com/browse/MTV-324[(MTV-324)]

Reviewer: ```suggestion
When a user specifies an incorrect password for RHV providers, he no longer gets locked in RHV. An error returns when the RHV manager is accessible when adding the provider. If the RHV manager is not accessible, the provider is added, but there would be no further attempt after failing, due to incorrect credentials. link:https://issues.redhat.com/browse/MTV-324[(MTV-324)]
```
?

.Convert i440fx to q35

> BIOS of i440fx chipset is reverted to BIOS of q35. link:https://issues.redhat.com/browse/MTV-430[(MTV-430)]

Reviewer: `BIOS of i440fx chipset is reverted to BIOS of q35` doesn't seem accurate, how about: `Migrations of virtual machines with i440fx chipset is now supported. The chipset is converted to q35 during the migration.`

Author: Changed to your suggestion

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-24_{context}"]
= {project-full} 2.4

You can migrate virtual machines (VMs) from VMware vSphere or {rhv-full} or {osp} to {virt} with {the-lc} {project-first}.

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-24_{context}"]
== Technical changes

This release has the following technical changes:

> .OpenStack migration

Reviewer: @anarnold97 sorry, I wasn't specific about "section" before - what i meant is to have it as the first item in the  "New features and enhancements" section below

Author: @ahadas thanks for all your help. It is tough picking up something like this half way through

The migrated virtual machines are no longer scheduled on the target OCP cluster. This enables migrating VMs that cannot start due to limit constraints at the migration time on the target.

.StorageProfile resource needs to be updated for a non-provisioner storage class

You must update the `StorageProfile` resource with `accessModes` and `volumeMode` for non-provisioner storage classes such as NFS. The documentation includes a link to the relevant procedure.

[id="new-features-and-enhancements-24_{context}"]
== New features and enhancements

This release has the following features and improvements:

.OpenStack migration

OpenStack is added as a source provider. Only cold migrations are supported as tech preview in this release.

.OCP console plugin

This version of {project-full} has replaced the stand-alone UI with a UI that is integrated in the OCP Console. The new UI operates as an OCP Console plugin that adds the sub-menu `Migration` to the navigation bar. It is implemented in version 2.4, disabling the old UI. You can enable the old UI by setting `feature_ui: true` in ForkliftController. link:https://issues.redhat.com/browse/MTV-427[(MTV-427)]

.Skip certification option

> Skip certificate validation' option was added to VMware and RHV providers. If checked, the provider's certificate will not be validated and the UI will not ask for specifying a CA certificate.

Reviewer 2: ```suggestion
'Skip certificate validation' option was added to VMware and RHV providers. If checked, the provider's certificate will not be validated and the UI will not ask for specifying a CA certificate.
```

Author: Changed as requested

@quiyuann resolved

.Only third-party certificate required

When defining a RHV provider that set with third-party certificate, only the third-party certificate needs to be specified and MTV would retrieve the Manager CA certificate automatically.

.Conversion of VMs with RHEL9 guest operating system

> "For cold migrations from vSphere to the local OCP cluster, we use virt-v2v on RHEL 9 that supports conversion of virtual machines with RHEL 9 as the guest operating system". link:https://issues.redhat.com/browse/MTV-332[(MTV-332)]

Reviewer: ```suggestion
For cold migrations from vSphere to the local OCP cluster, we use virt-v2v on RHEL 9 that supports conversion of virtual machines with RHEL 9 as the guest operating system. link:https://issues.redhat.com/browse/MTV-332[(MTV-332)]
```

.Unclear error status message for VM with no operating system

The error status message for a VM with no operating system on the *Migration plan details* page of the web console does not describe the reason for the failure. link:https://bugzilla.redhat.com/show_bug.cgi?id=2008846[(BZ#22008846)]

.Log archive file includes logs of a deleted migration plan or VM

If you delete a migration plan and then run a new migration plan with the same name or if you delete a migrated VM and then remigrate the source VM, the log archive file created by the {project-short} web console might include the logs of the deleted migration plan or VM. link:https://bugzilla.redhat.com/show_bug.cgi?id=2023764[(BZ#2023764)]

.QEMU guest agent is not installed on migrated VMs

The QEMU guest agent is not installed on migrated VMs. Workaround: Install the QEMU guest agent with a post-migration hook. link:https://bugzilla.redhat.com/show_bug.cgi?id=2018062[(BZ#2018062)]

.Migration of virtual machines with encrypted partitions fails during conversion

Occurs for vSphere: Migrations from RHV and OpenStack don't fail, but the encryption key may be missing on the target OCP cluster.

> .RHV snapshots are not deleted after a successful migration

Reviewer 2: @ahadas , do we need to also state that VMWare snapshots are not deleted after a successful migration?

Reviewer: yeah, but for openstack is should be a bit different:
```
.Snapshots that are created during the migration in OpenStack are not deleted
The Migration Controller service does not delete snapshots that are created during the migration for source virtual machines in OpenStack automatically. Workaround: the snapshots can be removed manually on OpenStack. 
```

> The Migration Controller service does not delete snapshots automatically after a successful warm migration of a {rhv-short} VM. You can link:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.9/html-single/virtualization/index#virt-deleting-vm-snapshot-web_virt-managing-vm-snapshots[delete a virtual machine snapshot manually]. link:https://bugzilla.redhat.com/show_bug.cgi?id=2053183[(BZ#2053183)] link:https://issues.redhat.com/browse/MTV-349[(MTV-349)]

Reviewer: looking at this link - it describes how to remove snapshots in Openshift virtualization while the problem is with snapshots that remain on RHV. I'd drop this link and just say that the snapshots can be removed from RHV instead

.OCP console plugin

The {product-full} Operator now integrates the {product-short} web console into the {ocp} web console. The new UI operates as an OCP Console plugin that adds the sub-menu `Migration` to the navigation bar. It is implemented in version 2.4, disabling the old UI. You can enable the old UI by setting `feature_ui: true` in ForkliftController. link:https://issues.redhat.com/browse/MTV-427[(MTV-427)]

.Skip certification option

'Skip certificate validation' option was added to VMware and RHV providers. If selected, the provider's certificate will not be validated and the UI will not ask for specifying a CA certificate.

.Only third-party certificate required

Only the third-party certificate needs to be specified when defining a RHV provider that sets with the Manager CA certificate.

.Conversion of VMs with RHEL9 guest operating system

Cold migrations from vSphere to a local Red Hat OpenShift cluster use virt-v2v on RHEL 9. link:https://issues.redhat.com/browse/MTV-332[(MTV-332)]

[id="known-issues-24_{context}"]

> == Known issues

Reviewer: Another known issue:
```
.Selection of migration transfer network does not apply to cold migrations to the local {ocp} instance

Cold migrations from vSphere, {rhv-full} or {osp} to the {ocp} instance that {project-full} runs on, does not necessarily transfer the disks on the selected migration transfer network.
```

Author: @ahadas ACK - thanks

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

[id="rn-24_{context}"]
= {project-full} 2.4

Migrate virtual machines (VMs) from VMware vSphere or {rhv-full} or {osp} to {virt} with {the-lc} {project-first}.

The release notes describe technical changes, new features and enhancements, and known issues.

[id="technical-changes-24_{context}"]
== Technical changes

This release has the following technical changes:

.Faster disk image migration from RHV

> Disk images are not converted anymore using virt-v2v when migrating from RHV. This way, the migration is faster and not limited to guest operating systems that are supported by virt-v2v. link:https://github.com/kubev2v/forklift-controller/issues/403[(forklift-controller#403)]

Reviewer 3: Consider: 1. "Disk images are no longer converted..."
2. "This change makes migrations faster and also allows migration for guest operating systems that are not supported by virt-vsv" or "As a result..." or some such.
Required: backticks for virt-v2v

.Faster disk transfers by ovirt-imageio client (ovirt-img)

> Disk transfers use ovirt-imageio client (ovirt-img) instead of Containerized Data Import (CDI) when migrating from RHV to the local OCP cluster, accelerating the migration.

Reviewer 3: 1. backticks
2. OCP is not used in customer-facing documentation -- use either Red Hat OpenShift or OpenShift Container Platform if you mean the that specifically

.Faster migration using conversion pod disk transfer

When migrating from vSphere to the local OCP cluster, the conversion pod transfers the disk data instead of Containerized Data Importer (CDI), accelerating the migration.

.Migrated virtual machines are not scheduled on the target OCP cluster

The migrated virtual machines are no longer scheduled on the target OCP cluster. This enables migrating VMs that cannot start due to limit constraints at the migration time on the target.

.StorageProfile resource needs to be updated for a non-provisioner storage class

You must update the `StorageProfile` resource with `accessModes` and `volumeMode` for non-provisioner storage classes such as NFS. The documentation includes a link to the relevant procedure.

> 

Reviewer 3: Why do we need to say that the documentation tells us how to do this? Not part of of other item's description and it's not part of our description of _new features_. If you're trying to warn the user that we only provide a link, rather than full instructions...there's no need for that.

[id="new-features-and-enhancements-24_{context}"]
== New features and enhancements

This release has the following features and improvements:

> 

Reviewer 3: Everything in the section above except "StorageProfile" is an improvement. Please move the relevant items to this section. I'm not thrilled by this division of topics (technical changes vs new features), but that's the existing structure, so let's go with it.

.OpenStack migration

OpenStack is added as a source provider. Only cold migrations are supported as tech preview in this release.

> 

Reviewer 3: {product-short} now supports migrations with {scp} as a source provider. This feature is a provided as a Technology Preview and only supports cold migrations.

Add Tech Preview boilerplate (swipe from  https://github.com/kubev2v/forklift-documentation/pull/362/files#diff-0e16c9b5e3ce0094b45256588681d4c88f889be367c58897e03feeb8d04a1b19)

.OCP console plugin

> This version of {project-full} has replaced the stand-alone UI with a UI that is integrated in the OCP Console. The new UI operates as an OCP Console plugin that adds the sub-menu `Migration` to the navigation bar. It is implemented in version 2.4, disabling the old UI. You can enable the old UI by setting `feature_ui: true` in ForkliftController. link:https://issues.redhat.com/browse/MTV-427[(MTV-427)]

Reviewer 3: The {product-full} Operator now integrates the {product-short} web console into the {ocp} web console. 

That's it. Consult with Arik if we want to discuss enabling the old UI -- we dropped that section from the guide.

.Skip certification option

'Skip certificate validation' option was added to VMware and RHV providers. If checked, the provider's certificate will not be validated and the UI will not ask for specifying a CA certificate.

> 

Reviewer 3: The...option.
"Selected," not "checked". 
If selected, the provider's CA certificate is not validated.  Why discuss what the UI does when the option is selected? Only add this if you are concerned veteran users will be concerned that they don't see the option AFTER they chose not to use it (Not likely...).

.Only third-party certificate required

When defining a RHV provider that set with third-party certificate, only the third-party certificate needs to be specified and MTV would retrieve the Manager CA certificate automatically.

> 

Reviewer 3: Fix the grammar and syntax. Hint: add articles.

.Conversion of VMs with RHEL9 guest operating system

For cold migrations from vSphere to the local OCP cluster, we use virt-v2v on RHEL 9 that supports conversion of virtual machines with RHEL 9 as the guest operating system. link:https://issues.redhat.com/browse/MTV-332[(MTV-332)]

[id="known-issues-24_{context}"]
== Known issues

This release has the following known issues:

.Deleting migration plan does not remove temporary resources

Deleting a migration plan does not remove temporary resources such as importer pods, conversion pods, config maps, secrets, failed VMs and data volumes. +
You must archive a migration plan before deleting it in order to clean up the temporary resources. link:https://bugzilla.redhat.com/show_bug.cgi?id=2018974[(BZ#2018974)]

> 

Reviewer 3: Make sure the '+" sign is not visible in the preview. It should be a separate line in the file.

.Unclear error status message for VM with no operating system

The error status message for a VM with no operating system on the *Migration plan details* page of the web console does not describe the reason for the failure. link:https://bugzilla.redhat.com/show_bug.cgi?id=2008846[(BZ#22008846)]

> 

Reviewer 3: The page is now called Plans -- either change your text or or remove the bold so your text tells the user what they are looking for without specifying its exact title in the UI. Never rely on Jiras for UI details, especially when you can check the UI yourself.

.Log archive file includes logs of a deleted migration plan or VM

If deleting a migration plan and then running a new migration plan with the same name or if deleting a migrated VM and then remigrate the source VM, the log archive file created by the {project-short} web console might include the logs of the deleted migration plan or VM. link:https://bugzilla.redhat.com/show_bug.cgi?id=2023764[(BZ#2023764)]

.QEMU guest agent is not installed on migrated VMs

The QEMU guest agent is not installed on migrated VMs. Workaround: Install the QEMU guest agent with a post-migration hook. link:https://bugzilla.redhat.com/show_bug.cgi?id=2018062[(BZ#2018062)]

> 

Reviewer 3: Good!

.Migration of virtual machines with encrypted partitions fails during conversion

Occurs for vSphere: Migrations from RHV and OpenStack don't fail, but the encryption key may be missing on the target OCP cluster.

> 

Reviewer 3: Consider: "vSphere only: Migrations from ... do not fail...".
Try to use the attributes as often as you can -- that way if we change the meaning of one, as we have with {ocp}, all you have to do is make one change in the attributes file. The attributes file has different names in different repos. Inthe MTV repo, it's called _common-attributes.adoc_ .

.Snapshots that are created during the migration in OpenStack are not deleted

The Migration Controller service does not delete snapshots that are created during the migration for source virtual machines in OpenStack automatically. Workaround: the snapshots can be removed manually on OpenStack. 

> 

Reviewer 3: Good description. If there is a BZ or Jira, please add it.

Warm migration from RHV fails if a snapshot operation is performed on the source VM. If the user performs a snapshot operation on the source VM at the time when a migration snapshot is scheduled, the migration fails instead of waiting for the user’s snapshot operation to finish. link:https://bugzilla.redhat.com/show_bug.cgi?id=2057459[(BZ#2057459)]

.Cannot schedule migrated VM with multiple disks to more than one storage classes of type hostpath

When migrating a VM with multiple disks to more than one storage classes of type hostpath, it may result in a VM that cannot be scheduled. Workaround: It is recommended to use shared storage on the target OCP cluster.

.Deleting migrated VM does not remove PVC and PV

When removing a VM that was migrated, its PVCs and PVs are not deleted. +
Workaround: remove the CDI importer pods and then remove the remaining PVCs and PVs. link:https://issues.redhat.com/browse/MTV-492[(MTV-492)]

> 

Reviewer 3: Again, I'm not sure why there's a plus + here. To keep two paragraphs together, enter a plus on the line between them in the .adoc file:
Paragraph A
+
Paragraph B

.PVC deletion hangs after archiving and deleting migration plan

When archiving and deleting a migration plan that failed, its PVCs and PVs are not deleted. Workaround: remove the CDI importer pods and then remove the remaining PVCs and PVs. link:https://issues.redhat.com/browse/MTV-493[(MTV-493)]

> 

Reviewer 3: Consider: "When a migration fails, its PVCs and PVs are not deleted as expected when its migration plan is archived and deleted. Workaround..."

.VM with multiple disks may boot from non-bootable disk after migration

VM with multiple disks that was migrated may not be able to boot on the target OCP cluster. +

> Workaround: set the boot order appropriately to boot from the bootable disk. link:https://issues.redhat.com/browse/MTV-433[(MTV-433)]

Reviewer 3: The IBM style guide uses "might" instead of "may" to remove possible ambiguity.  Check the + sign.
Consider:  An uppercase letter following a colon, especially when starting a complete sentence.

.Non-supported guest operating systems in warm migrations

Warm migrations and migrations to remote OCP clusters from vSphere do not support all types of guest operating systems that are supported in cold migrations to the local OCP cluster. It is a consequence of using RHEL 8 in the former case and RHEL 9 in the latter case. +
See link:https://access.redhat.com/articles/1351473[Converting virtual machines from other hypervisors to KVM with virt-v2v in RHEL 7, RHEL 8, and RHEL 9] for the list of supported guest operating systems.

.VMs from vSphere with RHEL 9 guest operating system may start with network interfaces that are down

When migrating VMs that are installed with RHEL 9 as guest operating system from vSphere, their network interfaces could be disabled when they start in OpenShift Virtualization. link:https://issues.redhat.com/browse/MTV-491[(MTV-491)]

[id="resolved-issues-24_{context}"]
== Resolved issues

This release has the following resolved issue:

.Improve invalid/conflicting VM name handling

> Improve the automatic renaming of VMs during migration to fit RFC 1123. This feature that was introduced in 2.3.4 is enhanced to cover more special cases. link:https://issues.redhat.com/browse/MTV-212[(MTV-212)]

Reviewer 3: Please state the issue that was resolved. It sounds like something that worked before is now better.

.Prevent locking user accounts due to incorrect credentials

If a user specifies an incorrect password for RHV providers, they are no longer locked in RHV. An error returns when the RHV manager is accessible and adding the provider. If the RHV manager is not accessible, the provider is added, but there would be no further attempt after failing, due to incorrect credentials. link:https://issues.redhat.com/browse/MTV-324[(MTV-324)]

.Users without cluster-admin role can create new providers
Previously, the `cluster-admin` role was required to browse and create providers. In this release, users with sufficient permissions on MTV resources (providers, plans, migrations, networkmaps, storagemaps, hooks) can operate MTV without cluster-admin permissions. link:https://issues.redhat.com/browse/MTV-334[(MTV-334)]

> 

Reviewer 3: Is this a fix or an improvement?

.Convert i440fx to q35

> 

Reviewer 3: This might be a technical change or a new feature. It doesn't read like a problem that was resolved.

Warm migrations and migrations to remote OCP clusters from vSphere do not support all types of guest operating systems that are supported in cold migrations to the local OCP cluster. It is a consequence of using RHEL 8 in the former case and RHEL 9 in the latter case. +
See link:https://access.redhat.com/articles/1351473[Converting virtual machines from other hypervisors to KVM with virt-v2v in RHEL 7, RHEL 8, and RHEL 9] for the list of supported guest operating systems.

.VMs from vSphere with RHEL 9 guest operating system may start with network interfaces that are down

> When migrating VMs that are installed with RHEL 9 as guest operating system from vSphere, their network interfaces could be disabled when they start in OpenShift Virtualization

Author: I will update with the ticket number when we get one from QE

Author: thanks
